docarray/document/mixins/multimodal.py

```diff
@@ -139,3 +139,28 @@ def _from_obj(cls, obj, obj_type, field) -> typing.Tuple['Document', AttributeTy
         else:
             raise ValueError(f'Unsupported type annotation')
         return doc, attribute_type
+
+    def __getattr__(self, attr):
+        has_metadata = (
+            super().__getattribute__('_data') is not None
+            and getattr(self._data, '_metadata') is not None
+        )
+        if (
+            has_metadata
+            and self.is_multimodal
+            and attr in self._metadata['multi_modal_schema']
+        ):
+            mm_attr_da = self.get_multi_modal_attribute(attr)
+            collected_attributes = []
+            for d in mm_attr_da:
+                if d.is_multimodal:  # comes from nested dataclass, add Document as is
+                    collected_attributes.append(d)
+                else:  # comes from a non-nested attribute, add the value
+                    collected_attributes.append(d.content)
+            return (
+                collected_attributes
```

Comment:
can you point me to an example where the List would apply and when a single element would apply?

Reply:
You can see it in the test [here](https://github.com/jina-ai/docarray/pull/425/files#diff-daea3d13c630b543a87c4e19c4607effce9121a41c629bb4297b892070c56692R644-R648), if the type is `List[T]` then it will be a list, otherwise a single element. Same applies for nested dataclasses, in that case the return will be a (list of) document